File: WarehouseManagementWeb.Application/Dto/Input/Resource/ChangeStatusResourceInput.cs
using WarehouseManagementWeb.Domain.Enums;

namespace WarehouseManagementWeb.Application.Dto.Input.Resource
{
    /// <summary>
    /// Класс входной модели обновления статуса ресурса.
    /// </summary>
    public class ChangeStatusResourceInput
    {
        /// <summary>
        /// Id ресурса.
        /// </summary>
        public int ResourceId { get; set; }

        /// <summary>
        /// Статус ресурса.
        /// </summary>
        public DirectoryStatusEnum ResourceStatusEnum { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/directory/resource/models/input/change-status-resource-input.ts
import { DirectoryStatusEnum } from '../../../enums/directory-status-enum';

/**
 * Класс входной модели обновления статуса ресурса.
 */
export class ChangeStatusResourceInput {
  /**
   *  Id ресурса.
   */
  resourceId: number = 0;

  /**
   * Статус ресурса.
   */
  resourceStatusEnum: DirectoryStatusEnum = 0;
}
